src/libs/sfdk/signingutils.cpp

```diff
@@ -0,0 +1,374 @@
+
+    auto runner = std::make_unique<GpgRunner>(arguments);
+    if (runner->process()->program().isEmpty()) {
+        m_lastError = GPG_NOT_FOUND_ERROR;
+        QTimer::singleShot(0, context, std::bind(functor, false, GpgKeyInfoList()));
+        return;
+    }
+
+    QObject::connect(runner.get(), &GpgRunner::done, context, [this, functor, process = runner->process()](bool ok) {
+         GpgKeyInfoList keysInfo;
+         if (ok)
+             keysInfo = gpgKeyInfoListFromOutput(process->readAllStandardOutput().trimmed());
+         else
+             m_lastError = QString::fromUtf8(process->readAllStandardError());
+
+         functor(ok, keysInfo);
+     });
+    commandQueue()->enqueue(std::move(runner));
+}
+
+void SigningUtilsPrivate::exportSecretKey(const QString &fingerprint,
+                                          const Utils::FilePath &passphraseFile,
+                                          const Utils::FilePath &outputPath,
+                                          const QObject *context,
+                                          const Functor<bool> &functor)
+{
+    if (fingerprint.isEmpty()) {
+        m_lastError = "Failed to export secret key: empty fingerprint";
+        QTimer::singleShot(0, context, std::bind(functor, false));
+        return;
+    }
+
+    if (!passphraseFile.exists() && !passphraseFile.isEmpty()) {
+        m_lastError = "Failed to export secret key: passphrase file set but doesn't exit";
+        QTimer::singleShot(0, context, std::bind(functor, false));
+        return;
+    }
+
+    //doesn't export if exists
+    if (outputPath.exists()) {
+        QTimer::singleShot(0, context, std::bind(functor, true));
+        return;
+    }
+
+    bool passphraseValid;
+    execAsynchronous(std::tie(passphraseValid), std::mem_fn(&SigningUtilsPrivate::verifyPassphrase), this,
+            fingerprint, passphraseFile.toString());
+
+    if (!passphraseValid) {
+        QTimer::singleShot(0, context, std::bind(functor, false));
+        return;
+    }
+
+    if (!outputPath.parentDir().exists()) {
+        QDir dir(outputPath.parentDir().toString());
+        if (!dir.mkdir(dir.absolutePath())) {
+            m_lastError = "Failed to export secret key: failed to create folder for keys: "
+                          + dir.absolutePath();
+            QTimer::singleShot(0, context, std::bind(functor, false));
+            return;
+        }
+    }
+
+    QStringList arguments;
+    arguments << "--batch"
+              << "--output" << outputPath.toString() << "--pinentry-mode=loopback";
+
+    if (!passphraseFile.isEmpty())
+        arguments << "--passphrase-file" << passphraseFile.toString();
+
+    arguments << "--export-secret-keys" << fingerprint;
+
+    auto runner = std::make_unique<GpgRunner>(arguments);
+    if (runner->process()->program().isEmpty()) {
+        m_lastError = GPG_NOT_FOUND_ERROR;
+        QTimer::singleShot(0, context, std::bind(functor, false));
+        return;
+    }

```

Comment:
This is too late to check for this. We should never enter this function if gpg was not found. All GPG related UIs should be disabled if GPG executable is not available. Simply drop this and the other occurrences.